plugin/src/main/java/org/opensearch/ml/action/mcpserver/OpenSearchMcpServerTransportProvider.java

```diff
@@ -121,10 +121,10 @@ public Mono<HttpChunk> handleSseConnection(StreamingRestChannel channel, boolean
                     // Send initial endpoint event
                     log.debug("Sending initial endpoint event to session: {}", sessionId);
                     String result;
-                    if (ssePrefix) {
-                        result = String.format(Locale.ROOT, "/sse/message?sessionId=%s", sessionId);
+                    if (appendToBaseUrl) {
+                        result = String.format(Locale.ROOT, "/_plugins/_ml/mcp/sse/message?sessionId=%s", sessionId);
                     } else {
-                        result = String.format(Locale.ROOT, "/message?sessionId=%s", sessionId);
+                        result = String.format(Locale.ROOT, "/sse/message?sessionId=%s", sessionId);
```

Comment:
Please create an issue about adding corresponding missing unit test for these classes. I'm approving for now to unblock 3.0 GA.